### PR TITLE
feat: allow ignoring value in comparison

### PR DESCRIPTION
**Pull request by eduardo-matos**
There are times when you don't care what value you're comparing to.

Let's say an API returns the following response:

```js
[{ id: 8873516, name: "John", age: 35 }]
```

And you care about only the name and age. Then you could do something like this:

```js
deepEqual(apiResponse, [{ id: deepEqual.ANY, name: 'John', age: 35 }]) // true;
```

This feature is heavily influenced by Python's mocking library [ANY](https://docs.python.org/3/library/unittest.mock.html#any).

### CODE REVIEW

**Comment by lgmontenegro** [comment]
Sure!
I support the idea of validating only the data that matters for the scoped test.

**Comment by keithamus** [comment]
Thanks for the PR @eduardo-matos! I feel like this should probably be done at a higher level though; with the custom `comparator` function (as per code docs): https://github.com/chaijs/deep-eql/blob/04d6da6518f8ddc288638ca42503752028810120/index.js#L82-L94

**Comment by eduardo-matos** [comment]
@keithamus Do you mean creating a function called `anyEqual` (code below) and executing it before [`simpleEqual`](https://github.com/chaijs/deep-eql/blob/04d6da6518f8ddc288638ca42503752028810120/index.js#L100-L103) or are you suggesting implementing this comparator on each project that needs this functionality?

```js
function anyEqual(leftHandOperand, rightHandOperand) {
    if (leftHandOperand === ANY || rightHandOperand === ANY) {
        return true;
    }
}


```

**Comment by lucasfcosta** [comment]
Hi @eduardo-matos, thanks for your PR.

By seeing this PR and also maintaining [Sinon](https://github.com/sinonjs/sinon) I see more necessity of having [`matchers`](http://sinonjs.org/docs/#matchers)

I think this is a great opportunity for a Plugin, instead of adding it to the core since it introduces/encourages non-determinism.

Also, since we're just ignoring a key, I think we would benefit more from adding the possibility of ignoring a key by its name at all.

This would also allow us to easily tackle issues that address that (ignoring a key in `deepEqual`) at the Chai main repo.

tl;dr I think this feature is necessary but I think it should be implemented in another way, which is allowing users to list the keys to ignore instead of using what looks like a "matcher".

**Comment by eduardo-matos** [comment]
@lucasfcosta I agree with you. A plugin will fit much better than changing chai's core (at least for this feature).

This PR may be closed in my opinion.

**Comment by lucasfcosta** [comment]
Thanks for sharing your thoughts @eduardo-matos.

If you need any help with the plugin or if you'd like to implement a feature that ignores keys into `deep-eql` I'll be happy to help.

PS.: I'm Brazilian too so if you want to speak some Portuguese and do pair programming I'm open to that 😄

**Comment by eduardo-matos** [comment]
@lucasfcosta 
Actually I've just released a Chai plugin that does what I proposed in this PR.  
If you have any thoughts, please let me know.

https://github.com/eduardo-matos/chai-any-eql

Thanks!
